src/main/java/org/jitsi/jigasi/transcription/TranslationManager.java

```diff
@@ -0,0 +1,160 @@
+     */
+    public TranslationManager(TranslationService service)
+    {
+        translationService = service;
+    }
+
+    /**
+     * Adds a {@link TranslationResultListener} to the list of
+     * listeners to be notified of a TranslationResult.
+     *
+     * @param listener to be added.
+     */
+    public void addListener(TranslationResultListener listener)
+    {
+        listeners.add(listener);
+    }
+
+    /**
+     * Adds a language tag to the list of target languages for
+     * translation or increments its count in the map if key exists.
+     *
+     * @param language to be added to the list
+     */
+    public void addLanguage(String language)
+    {
+        if(language == null || language.isEmpty())
+            return;
+
+        languages.put(language, languages.getOrDefault(language, 0) + 1);
+    }
+
+    /**
+     * Decrements the language count in the map and removes the language if
+     * no more participants need it.
+     *
+     * @param language whose count is to be decremented
+     */
+    public void removeLanguage(String language)
+    {
+        if(language == null)
+            return;
+
+        int count = languages.get(language);
+
+        if (count == 1)
+        {
+            languages.remove(language);
+        }
+        else
+        {
+            languages.put(language, count - 1);
+        }
+    }
+
+    /**
+     * Translates the received {@link TranscriptionResult} into required languages
+     * and returns a list of {@link TranslationResult}s.
+     *
+     * @param result the TranscriptionResult notified to the TranslationManager
+     * @return list of TranslationResults
+     */
+    public ArrayList<TranslationResult> getTranslations(
+        TranscriptionResult result)
+    {
+        ArrayList<TranslationResult> translatedResults
+            = new ArrayList<>();
+
+        for(String language : languages.keySet())
+        {
+            String translatedText = translationService.translate(
+                result.getAlternatives().iterator().next().getTranscription(),
+                result.getLanguage().substring(0,2),
+                language);
+
+            translatedResults.add(new TranslationResult(
+                result,
+                language,
+                translatedText));
+        }
+        return translatedResults;
+    }
+
+    /**
+     * Translates the received {@link TranscriptionResult} into the
+     * target languages and notifies the {@link TranslationResultListener}'s
+     * of the {@link TranslationResult}s.
+     *
+     * @param result the result which has come in.
+     */
+    @Override
+    public void notify(TranscriptionResult result)
+    {
+        if(!result.isInterim())
+        {
+            ArrayList<TranslationResult> translations
+                = getTranslations(result);
+            for (TranslationResultListener listener : listeners)
+            {
+                for (TranslationResult translation : translations)
```

Comment:
` translations.forEach(listener::notify);` might be more elegant. Or even
`listeners.forEach(l -> translations.forEach(l::notify));`